### DISCUSSION TITLE
Why does the share-manager crashes ?

### DISCUSSION THREAD

**Discussion by emmanuelW935** [longhorn/longhorn#4889] [Q&A]
## Question
Longhorn 1.3.2, RWX, single node, k3s 1.24.7, rhel 8.6: 
Why does the share manager crashes ? In which direction to look ? 

When using long horn in a single node with a couple of pods to test it, the share-manager crashes quickly when files are created and copied.

The use case is: 

using the attached storage class  (1 replica ).
using the attached pvc/pv and couple of pods.

with the pv and copy of files into a shell of the pod, it crashes the sharemanager after some cp.
copy of files using kubectl cp to the pod does the same.

## Environment

 - Longhorn version:	1.3.2
 - Kubernetes version:	v1.24.7+k3s1
 - Node config
   - OS type and version rhel 8.6
   - CPU per node: 32
   - Memory per node: 62 Gb
   - Disk type ( xfs )
/dev/mapper/rhel-root xfs       
/dev/mapper/rhel-home xfs       
/dev/sda1             xfs      
   - Network bandwidth and latency between the nodes: not applicable.

 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal):
DryIce

## Additional context
-nfs client is ok.

-Event Readiness probe failed: cat: /var/run/ganesha.pid: No such file or directory is observed when the share manager crashes.
-describe node is attached.
-there is a lot of space remaining on the volume.
-size of the files is around 1Gb each.
Also for the env. check there are no issues reported:
[INFO]  Required dependencies are installed.
[INFO]  Waiting for longhorn-environment-check pods to become ready (0/1)...
[INFO]  All longhorn-environment-check pods are ready (1/1).
[INFO]  Required packages are installed.
[INFO]  MountPropagation is enabled.
[INFO]  Cleaning up longhorn-environment-check pods...
[INFO]  Cleanup completed.

[node.txt](https://github.com/longhorn/longhorn/files/10022348/node.txt)
[storage.yaml.txt](https://github.com/longhorn/longhorn/files/10022360/storage.yaml.txt)
[lhpodsample.yaml.txt](https://github.com/longhorn/longhorn/files/10022362/lhpodsample.yaml.txt)

**Comment by shuo-wu** [comment]
If possible, can you try to use a RWO volume or a RWX volume with a single pod instead?
The share manager pod killing maybe caused by the underlying volume engine crash.  And the CPU resource exhaustion can be one possible reason.
For future investigation, you may need to provide the support bundle. The info you already provided does not contain any clue.

**Reply by emmanuelW935** [reply]
This can be reproduced with a single pod + rwx ( the support bundle is attached ), it allows to see some R/W time outs ( also there were no CPU/Disk/Memory exhaustion during the test ).
[longhorn-support-bundle_f326eaf1-1eab-4aba-83ca-a0a310c5c908_2022-11-18T15-19-50Z.zip](https://github.com/longhorn/longhorn/files/10055706/longhorn-support-bundle_f326eaf1-1eab-4aba-83ca-a0a310c5c908_2022-11-18T15-19-50Z.zip)
